lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -514,15 +533,33 @@ def post(self):
         are not in the class schema are ignored but logged.
         """
         json_data = request.get_json(silent=True)
+        if Auth.token_auth.current_user() is None:
+            if json_data and "user" in json_data:
+                abort(HTTPStatus.FORBIDDEN, message="Not Authorized")
         try:
             new_data = self.schema.validate(json_data)
+            # After the json data is validated make sure the user parameter is present
+            # in new_data for an endpoint which needs a "user" parameter.
+            # If the request does not contain the user field, it will be interpreted as
+            # a public dataaset query. Therefore after validating the json data we will
+            # manually add the user field as None to let the ensemle method understand that this
```

Comment:
"assemble" not "ensemble" ;-)

And note for other reviewers: I've already changed this in a forthcoming PR (which includes "publish" mechanism) so we won't have to inject a "user" key to satisfy the user term builder. So consider this a temporary bridge to slightly minimize my rebase conflicts!